easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -153,6 +182,124 @@ def build_step(self):
+
+    def python_safe_install(self, **kwargs):
+        """Install using 'python setup.py install --prefix', after verifying it does the right thing."""
+        cwd = os.getcwd()
+
+        # create dummy Python package to verify whether 'python setup.py install --prefix' does the right thing
+        tmpdir = tempfile.mkdtemp()
+        pkg = 'easybuild_pyinstalltest'
+        mkdir(os.path.join(tmpdir, pkg))
+        write_file(os.path.join(tmpdir, 'setup.py'), TEST_SETUP_PY % {'pkg': pkg})
+        test_py_script = '%s.py' % pkg
+        write_file(os.path.join(tmpdir, test_py_script), TEST_SCRIPT_PY)
+        test_data_file = '%s.txt' % pkg
+        write_file(os.path.join(tmpdir, test_data_file), 'data')
+        write_file(os.path.join(tmpdir, pkg, '__init__.py'), TEST_INIT_PY)
+
+        # install dummy Python package
+        try:
+            os.chdir(tmpdir)
+            testinstalldir = tempfile.mkdtemp()
+            self.python_install(prefix=testinstalldir)
+            os.chdir(cwd)
+        except OSError, err:
+            self.log.error("Failed to move to %s: %s" % (tmpdir, err))
+
+        # verify installation of dummy Python package
+        verified = True
+        full_pylibdir = os.path.join(testinstalldir, self.pylibdir)
+        cmds = [
+            ("python -c 'from %s import where; where()'" % pkg, full_pylibdir),
+            (test_py_script, testinstalldir),
+        ]
+        for cmd, out_prefix in cmds:
+            precmd = "PYTHONPATH=%s:$PYTHONPATH PATH=%s:$PATH" % (full_pylibdir, os.path.join(testinstalldir, 'bin'))
+            fullcmd = ' '.join([precmd, cmd])
+            (out, ec) = run_cmd(fullcmd, simple=False)
+            tup = (out_prefix, fullcmd)
+            if out.startswith(out_prefix):
+                self.log.debug("Found %s in output of '%s' during verification of dummy Python installation" % tup)
+            else:
+                tup = (tup[0], tup[1], ec, out)
+                self.log.warning("%s not found in output of '%s' (exit code: %s, output: %s)" % tup)
+                verified = False
+        pyver = get_software_version('Python')
+        if not pyver:
+            self.log.error("Python module not loaded.")
+        pyver = '.'.join(pyver.split('.')[:2])
+        datainstalldir = os.path.join(full_pylibdir, '%s-1.0-py%s.egg' % (pkg, pyver))
+        tup = (test_data_file, datainstalldir)
+        if os.path.exists(os.path.join(datainstalldir, test_data_file)):
+            self.log.debug("Found file %s in %s during verification of dummy Python installation" % tup)
+        else:
+            self.log.warning("Failed to find file %s in %s during verification of dummy Python installation" % tup)
+            verified = False
+
+        if verified:
+            self.log.debug("Verification of dummy Python installation OK.")
+        else:
+            self.log.error("Verification of dummy Python installation failed, setuptools not honoring --prefix?")
```

Comment:
no, setuptools may (apparently) blatently ignore `--prefix`

this seems to happen when setuptools itself was installed using `--user`